uid2_client/client.py

```diff
@@ -65,20 +64,37 @@ def refresh_keys(self):
             EncryptionKeysCollection containing the keys
         """
         req, nonce = self._make_v2_request(dt.datetime.now(tz=timezone.utc))
-        print(req)
-        resp = self._post('/v2/key/latest', headers=self._auth_headers(), data=req)
-        keys = [EncryptionKey(k['id'], k.get('site_id', -1), _make_dt(k['created']), _make_dt(k['activates']), _make_dt(k['expires']), base64.b64decode(k['secret']))
-            for k in json.loads(self._parse_v2_response(resp.read(), nonce)).get('body')]
-        return EncryptionKeysCollection(keys)
-
+        resp = self._post('/v2/key/sharing', headers=self._auth_headers(), data=req)
+        resp_body = json.loads(self._parse_v2_response(resp.read(), nonce)).get('body')
+        return self._parse_keys_json(resp_body)
+
+    def refresh_json(self, json_str):
+        body = json.loads(json_str)
+        return self._parse_keys_json(body['body'])
+
+
+    def _parse_keys_json(self, resp_body):
+        keys = []
+        for key in resp_body["keys"]:
+            keyset_id = None
+            if "keyset_id" in key:
+                keyset_id = key["keyset_id"]
+            key = EncryptionKey(key['id'],
+                                key.get('site_id', -1),
+                                _make_dt(key['created']),
+                                _make_dt(key['activates']),
+                                _make_dt(key['expires']),
+                                base64.b64decode(key['secret']),
+                                keyset_id)
+            keys.append(key)
+        return EncryptionKeysCollection(keys, resp_body["caller_site_id"], resp_body["master_keyset_id"],
+                                        resp_body["default_keyset_id"], resp_body["token_expiry_seconds"])
 
     def _make_url(self, path):
         return self._base_url + path
 
-
     def _auth_headers(self):
-        return {'Authorization': 'Bearer ' + self._auth_key}
-
+        return {'Authorization': 'Bearer ' + self._auth_key, "X-UID2-Client-Version": pkg_resources.get_distribution("uid2_client").version}
```

Comment:
what string would this add to the version header, "uid2_client-2.0.0" i think?
but ew wanna differentiate which language as well, so would be good to make it "uid-client-python-2.0.0"